### ISSUE TITLE
standardize core API functions

### ISSUE THREAD

**Issue by delgadom** [ClimateImpactLab/DataFS#55]
#### DataAPI
```
create_archive --> create
```

#### CLI

```
create_archive --> create
upload --> update
```

#### DataAPI

Add the folowing methods:
* `list()`
* `update(archive_name, filepath, **kwargs)`
* `metadata(archive_name)`
* `versions(archive_name)` 
* `download(archive_name, filepath)`

**Comment by jgerardsimcock** [comment]
What would `list()` do at the API level?

**Comment by delgadom** [comment]
same thing it does at the cli level ;)

I just want a consistent way of listing/filtering archives. And I'm concerned that we need a way of optimizing list_archive commands as we will soon have millions of archives. We should probably think of ways to limit the scope of these commands by default.

**Comment by delgadom** [comment]
Addressed in #84
